File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L6/04/Kasus7.ts
import { Scanner } from 'readline-sync';

function inputMatrix(matrix: number[][], n: number): void {
  const sc = new Scanner();

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      matrix[i][j] = sc.nextDouble();
    }
  }
}

function sumMajorDiagonal(matrix: number[][], n: number): number {
  if (n >= 0) {
    return matrix[n][n] + sumMajorDiagonal(matrix, --n);
  } else {
    return 0;
  }
}

function main(): void {
  const n = 4;
  console.log('Enter a 4 by 4 matrix row by row: ');
  const matrix: number[][] = new Array(n).fill(0).map(() => new Array(n).fill(0));

  inputMatrix(matrix, n);
  console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(matrix, n - 1)}`);
}

main();

File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L6/04/Kasus7.kt
package IR

import java.util.Scanner

/**
 *
 * @author C166AB8FBBBE55174E16A67530426A8A 19011F43894F06E2BA0CA74061427469
 */
fun inputMatrix(matrix: Array<DoubleArray>, n: Int) {
    val sc = Scanner(System.`in`)

    for (i in 0 until n) {
        for (j in 0 until n) {
            matrix[i][j] = sc.nextDouble()
        }
    }
}

fun sumMajorDiagonal(matrix: Array<DoubleArray>, n: Int): Double {
    return if (n >= 0) {
        matrix[n][n] + sumMajorDiagonal(matrix, n - 1)
    } else {
        0.0
    }
}

fun main(args: Array<String>) {
    val n = 4
    println("Enter a 4 by 4 matrix row by row: ")
    val matrix = Array(n) { DoubleArray(n) }

    inputMatrix(matrix, n)
    println("Sum of the elements in the major diagonal is ${sumMajorDiagonal(matrix, n - 1)}")
}

